tests/embedding_lookup.cc

```diff
@@ -0,0 +1,38 @@
+#include <vector>
+#include <iostream>
+
+#include "embeddings.h"
+#include "finalfusion.h"
+
+int main() {
+  Embeddings *embeddings;
```

Comment:
Don't use bare pointers. Use `unique_ptr` or a value type. Since there is no no-arg constructor and you want to wrap this in a `try` block, it's easiest to use one of these smart pointers. Saves you the `delete` later as well ;).